### ISSUE TITLE
flycheck-rust vs lsp-flycheck

### ISSUE THREAD

**Issue by TatriX** [flycheck/flycheck-rust#59]
I'm trying to understand should I use this project or use lsp-flycheck?

**Comment by fmdkdd** [comment]
`flycheck-rust` merely calls `cargo rustc`.  `lsp-flycheck` goes with `lsp-mode` and will interact with the Rust Language Server (RLS) using the language server protocol over stdio.

The latter may have faster turn-around, and other benefits (like auto-completion support), but may still be a bit rough around the edges.

If you just want errors/diagnostics in your buffer, use `flycheck-rust`.  If you want a more complete IDE experience, try `lsp-mode`/`lsp-rust` and see it if works for you.

**Comment by TatriX** [comment]
I see.
I have an issue with both packages: flycheck shows errors, well, rarely. Mostly I can get it work only once after I open the file. After editing and saving the file no errors will be shown.

**Comment by fmdkdd** [comment]
Are you, perhaps using rust nightly?  There is currently a bug (flycheck/flycheck#1381) that prevents errors to be shown with nightly.  Rust stable should be fine though, if not, that's another bug!

**Comment by TatriX** [comment]
Yeah, it's nightly.

**Comment by fmdkdd** [comment]
The fix for nightly has been merged.  Feel free to open a new issue if you encounter another bug in Flycheck :)
